internal/docker/docker.go

```diff
@@ -56,11 +56,13 @@ type Client struct {
 	logger               *zap.Logger
 }
 
-func NewDockerClient(config *Config, logger *zap.Logger) (*Client, error) {
+func NewDockerClient(config *Config, logger *zap.Logger, opts ...docker.Opt) (*Client, error) {
```

Comment:
I had to make this change to the internal docker client because like I said in the PR description, it was using the incorrect dialer.